### PR TITLE
apply changes incrementally

### PR DESCRIPTION
**Pull request by codinggosu**
**What this PR does / why we need it**:
loads changes incrementally, as described in issue #228 

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

This is a draft PR, just to make sure that this is the direction that this issue should be headed for.

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation**:

```docs

```

**Checklist**:
- [ ] Added relevant tests or not required
- [x] Didn't break anything

### CODE REVIEW

File: yorkie/packs/packs.go [hunk 1]
```diff
@@ -329,14 +331,21 @@ func pullSnapshot(
...
 
-	if err := doc.ApplyChangePack(change.NewPack(
-		docKey,
-		checkpoint.Initial.NextServerSeq(docInfo.ServerSeq),
-		changes,
-		nil,
-	)); err != nil {
-		return nil, nil, err
-	}
+	var startIdx, endIdx = 0, CHANGE_SEGMENT_SIZE;
```
Comment (codinggosu) [inline yorkie/packs/packs.go:334]: I haven't verified the logic of this, just wanted to make sure that this is indeed the right direction, before further implementation and testing

File: yorkie/packs/packs.go [hunk 2]
```diff
@@ -41,6 +41,8 @@ var (
 	ErrInvalidServerSeq = errors.New("invalid server seq")
 )
 
+const CHANGE_SEGMENT_SIZE = 100
```
[the remark this answers is not in the document]
Comment (hackerwins) [reply yorkie/packs/packs.go:44]: Please refer to the document below.
https://github.com/yorkie-team/yorkie/blob/main/CONTRIBUTING.md#code-style

**Comment by hackerwins** [comment]
Thank you for your contribution. I think the approach is good.

There are still many changes that need to be fetched from the DB. It seems that we need to include code to fetch from DB in the loop.

File: yorkie/packs/packs.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (codinggosu) [reply yorkie/packs/packs.go:44]: Thank you for the comment! As this was a draft PR mainly aimed at verifying the direction of the change, I didn't look too closely at the coding style. 
I will make sure to clean up all the code thoroughly and in accordance with the guide before the final change.

**Comment by codinggosu** [comment]
I will look over and test the code again with edge cases and edge values. 
After testing, I will mark the PR as ready!

File: yorkie/packs/packs.go [hunk 3]
```diff
@@ -316,26 +318,35 @@ func pullSnapshot(
...
+			changes,
+			nil,
+		)); err != nil {
+			return nil, nil, err
+		}
+		startIdx += changeSegmentSize
+		if endIdx == initialServerSeq {
+			break
+		}
 	}
```
Comment (hackerwins) [inline yorkie/packs/packs.go:350]: How about using a for loop a little more to shorten the code?

```suggestion
	for startIdx := snapshotInfo.ServerSeq + 1; startIdx < initialServerSeq; startIdx += changeSegmentSize {
		var endIdx uint64
		if startIdx+changeSegmentSize < initialServerSeq {
			endIdx = startIdx + changeSegmentSize
		} else {
			endIdx = initialServerSeq
		}

		changes, err := be.DB.FindChangeInfosBetweenServerSeqs(
			ctx,
			docInfo.ID,
			startIdx,
			endIdx,
		)
		if err != nil {
			return nil, nil, err
		}

		if err := doc.ApplyChangePack(change.NewPack(
			docKey,
			checkpoint.Initial.NextServerSeq(docInfo.ServerSeq),
			changes,
			nil,
		)); err != nil {
			return nil, nil, err
		}
	}
```

**Review by hackerwins — CHANGES_REQUESTED** [review]
Thanks for your contribution.

Is it possible to write test code for this code?

**Comment by codinggosu** [comment]
yes! I will add test code and update the pr

**Comment by codinggosu** [comment]
unfortunately, I can't find time to work on this issue anymore
I regret it, but it seems better to close this pr for now.

**Comment by hackerwins** [comment]
Good luck to you.
